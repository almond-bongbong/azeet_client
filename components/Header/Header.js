import React from 'react';
import Link from 'next/link';
import styled from 'styled-components';

const HeaderStyle = styled.div`
  & nav {
    
  }
  
  & button {
    display: block;
  }
`;

const Header = () => (
  <HeaderStyle>
    <nav>
      <Link href="/login"><a href="/login">Login</a></Link>
      <Link href="/about"><a href="/about">About</a></Link>
    </nav>
  </HeaderStyle>
);

export default Header;
