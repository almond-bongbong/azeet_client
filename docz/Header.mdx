---
name: Header
---

import { Playground, Props } from 'docz'
import Header from 'components/Header/Header'

# Header

<Props of={Header} />

## basic usage

<Playground>
  <Header />
</Playground>
