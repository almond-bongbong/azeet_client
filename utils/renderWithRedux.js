import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configStore from 'store/configure';
import Alert from 'components/Alert';
import Confirm from 'components/Confirm';
import Toast from 'components/Toast';

function renderWithRedux(ui, initialState) {
  const store = configStore(initialState);
  const utils = render(
    <Provider store={store}>
      {ui}
      <Alert />
      <Confirm />
      <Toast />
    </Provider>,
  );
  return {
    ...utils,
    store,
  };
}

export default renderWithRedux;
