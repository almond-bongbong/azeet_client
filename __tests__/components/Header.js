import React from 'react';
import renderWithRedux from 'utils/renderWithRedux';
import Header from 'components/Header/Header';

describe('Header', () => {
  const auth = { user: { id: 1, nickname: '길동' } };

  it('matches snapshot', () => {
    const utils = renderWithRedux(<Header />);
    expect(utils.container).toMatchSnapshot();
  });

  it('logged in user', () => {
    const { queryByText, getByAltText } = renderWithRedux(<Header />, { auth });
    const loginButton = queryByText('Login');
    getByAltText('thumbnail');
    expect(loginButton).toBeNull();
  });
});
