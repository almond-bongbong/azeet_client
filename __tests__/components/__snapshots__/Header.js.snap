// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <div
    class="Header__HeaderStyle-sc-8u78di-0 iCLpEf"
  >
    <h1>
      <a
        href="/"
      >
        LOGO
      </a>
    </h1>
    <nav>
      <a
        href="/login"
      >
        Login
      </a>
    </nav>
  </div>
</div>
`;
